runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcGetDataStream.java

```diff
@@ -68,10 +76,21 @@ final class GrpcGetDataStream
   private static final StreamingGetDataRequest HEALTH_CHECK_REQUEST =
       StreamingGetDataRequest.newBuilder().build();
 
-  /** @implNote {@link QueuedBatch} objects in the queue are is guarded by {@code this} */
+  static final FluentBackoff BACK_OFF_FACTORY =
+      FluentBackoff.DEFAULT
+          .withInitialBackoff(Duration.millis(10))
+          .withMaxBackoff(Duration.standardSeconds(10));
+
+  /**
+   * @implNote {@link QueuedBatch} objects in the queue should also be guarded by {@code this}.
+   *     Batches should be sent from the front of the queue and only removed from the queue once
+   *     added to the pending set of a physical stream.
+   */
+  @GuardedBy("this")
   private final Deque<QueuedBatch> batches;
 
-  private final Map<Long, AppendableInputStream> pending;
+  private final Supplier<Integer> batchesDebugSizeSupplier;
```

Comment:
thoughts on replacing this with the following?

`
 private final ConcurrentLinkedDeque<QueuedBatch> batches;

@SuppressWarnings("GuardedBy")
private int getBatchesSize() {
batches.size();
}`

Reply:
I prefer doing it without requiring a suppression since it doesn't seem too complicated.